src/libstore/store-api.cc

```diff
@@ -1358,14 +1358,14 @@ std::shared_ptr<Store> openFromNonUri(const std::string & uri, const Store::Para
                store in ~/.local/share/nix/root. */
             auto chrootStore = getDataDir() + "/nix/root";
             if (!pathExists(chrootStore)) {
+                warn("'%s' does not exist, so Nix will use '%s' as a chroot store", stateDir, chrootStore);
```

Comment:
Not sure the warning should be moved. That's misleading if an error occurs, because then it *doesn't* use a chroot store.